docs/report/report.tex

```diff
@@ -45,6 +45,92 @@ \subsubsection{Stretch Goals}
+Alternativ ginge es auch ohne Port-Forwarding mit nginx und ngrok \footnote{https://vatsalyagoel.com/setting-up-a-public-web-server-using-a-raspberry-pi-3/}.
+
+\subsubsection{URL für UI festlegen}
+Über ein DNS Service, der auf dem Raspberri Pi läuft, wäre es möglich für die statische Public-IP eine eigene URL anzulegen.
+Dafür sind verschiedene DNS Services möglich, potentiell ist auch eine Domain notwendig.
+
+\subsection{User Interface}
+\subsubsection{Zugriff auf NFC Reader im Cloud Szenario}
+Tendenziell sollte es nicht notwendig sein, dass der Raspberri Pi, der das Web-UI hosted, auf den jeweiligen NFC-Reader der Box zugreifen muss.
+Sollte das doch der Fall sein, könnte der Chip der Box ein API Call an den Raspberri Pi mit den NFC-Tag Daten machen, um das jeweilige hinterlegte Event zu triggern.
+
+\subsubsection{Login via Spotify, Youtube, etc.}
+Es gibt ein Feathers Plugin welches die Möglichkeit bietet OAuth Provider zu nutzen, um sich über andere Services wie Spotify anzumelden.
+
+\subsubsection{Gleichen Nutzer bei verschiedenen Loginvarianten wiedererkennen}
+Um gleiche Nutzer zu erkennen, müssten Merkmale angelegt werden, über die diese Nutzer wiedererkennbar wären.
+Die E-Mail wäre hierbei ein geeignetes Merkmal, das einzigartig sein sollte. Über Scopes in der OAuth Anfrage wird dieses in der Regel vom jeweiligen Provider mitgeliefert.
+Um die E-Mail als Merkmal zu verwenden, wird vorausgesetzt, dass Nutzer immer die gleiche E-Mail bei den unterschiedlichen Providern nutzen. Dies kann aber nicht immer gewährleistet werden und hängt stark von dem einzelnen Nutzer ab.
+Man könnte dem (bereits eingeloggten) Nutzer die Möglichkeit geben, weitere Accounts zu dem bestehenden hinzuzufügen, sodass ein Login über unterschiedliche Provider den gleichen Nutzer einloggen.
+Bei diesem Ansatz sollten dann die jeweiligen verbundenen Accounts zu einer unique User-ID verbunden werden.
+
+\subsubsection{Musik Artwork laden}
+Sollte bei der Verwendung von Spotify kein Problem sein, da zu jedem Titel / Lied eine Liste von Bildern enthalten sein sollte.\footnote{https://stackoverflow.com/questions/10123804/retrieve-cover-artwork-using-spotify-api}
+
+\subsubsection{Eigene Bilder hochladen}
+Eigene Bilder hochzuladen sollte möglich sein. In unserem Kontext mit Vue.js und Node.js würde das Plugin \textit{vue-picture-input} helfen.
+Mit einem Axios Post an das Backend könnte das Bild gesendet werden. \footnote{https://www.digitalocean.com/community/tutorials/vuejs-uploading-vue-picture-input}
+
+\subsubsection{Spotify Connect Lautsprecher auswählen}
+Das Auswählen von einem spezifischen Spotify Connect Lautsprecher ist möglich.
+Über einen API Call an die Spotify API mit dem Endpunkt \textit{/v1""/me""/player""/device} erhält man eine Liste von allen verbundenen Geräten, welche man dann dem Endpunkt zum Abspielen eines Liedes übergeben kann.
```

Comment:
'welche man dann' klingt auch etwas unschön

Reply:
adressed in commit ebddf4dd3e0ad75eae259bd72ee957a80ea9e62b